colorbleed/plugins/maya/publish/submit_switch_job.py

```diff
@@ -0,0 +1,108 @@
+import os
+import pprint
+
+from avalon import api
+from avalon.vendor import requests
+
+import pyblish.api
+
+
+def _get_script_dir():
+    """Get path to the image sequence script"""
+    try:
+        import colorbleed
+        scriptdir = os.path.dirname(colorbleed.__file__)
+        fusion_scripts = os.path.join(scriptdir,
+                                      "scripts",
+                                      "fusion")
+    except:
+        raise RuntimeError("This is a bug")
+
+    assert os.path.isdir(fusion_scripts), "Config is incomplete"
+    fusion_scripts = fusion_scripts.replace(os.sep, "/")
+
+    return fusion_scripts
+
+
+class SubmitDependentSwitchJobDeadline(pyblish.api.ContextPlugin):
+    """Run Switch Shot on specified comp as depending job
+
+    """
+
+    label = "Submit Switch Jobs to Deadline"
+    order = pyblish.api.IntegratorOrder + 0.2
+    hosts = ["maya"]
+    families = ["colorbleed.renderlayer"]
+
+    def process(self, context):
+
+        # Run it as depend on the last submitted instance
+        instance = context[-1]
+
+        AVALON_DEADLINE = api.Session.get("AVALON_DEADLINE",
+                                          "http://localhost:8082")
+        assert AVALON_DEADLINE, "Requires AVALON_DEADLINE"
+
+        job = instance.data.get("deadlineDependJob", None)
+        if not job:
+            self.log.warning("No dependent Job found")
+            return True
+
+        filepath = instance.data("flowFile", "")
+        if not filepath:
+            raise RuntimeError("No flow file (comp) chosen")
+
+        shot = api.Session["AVALON_ASSET"]
+        comment = instance.context.data["comment"]
+
+        scriptdir = _get_script_dir()
+        scriptfile = os.path.join(scriptdir, "deadline_swith_and_submit.py")
```

Comment:
Did you test this? :) Probably won't work because of a typo. The filename is: `deadline_switch_and_submit.py`

Reply:
> Saw an issues with retrieving the listedslaves, will fix this before merging

I saw this as well when attempting to fix the above mentioned issue